lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -743,6 +743,41 @@ OpFoldResult MuxOp::fold(ArrayRef<Attribute> constants) {
   return {};
 }
 
+void MuxOp::getCanonicalizationPatterns(OwningRewritePatternList &results,
+                                        MLIRContext *context) {
+  struct Folder final : public OpRewritePattern<MuxOp> {
+    using OpRewritePattern::OpRewritePattern;
+    LogicalResult matchAndRewrite(MuxOp op,
+                                  PatternRewriter &rewriter) const override {
+      auto width = op.getType().cast<IntegerType>().getWidth();
+
+      if (width == 1) {
+        APInt value;
+
+        // mux(a, true, b) -> or(a, b)
```

Comment:
Isn't this true for any width if the second argument is all ones?

Reply:
Nice generalization.  I think it would be something like:
`mux(a, allones, b)` -> `or(sext(a), b)`

For a single bit mux, the sext is a noop.  You can handle this with `createOrFold`.

Reply:
It looks `createOrFold` doesn't work since the creation of `sext` for 1 bit mux is rejected by the verification of `sext`. 

But anyway, I have generalized the width.